modules/core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/DocumentBlobManager.java

```diff
@@ -35,14 +35,27 @@
  */
 public interface DocumentBlobManager {
 
+    /**
+     * Reads a {@link Blob} from storage.
+     *
+     * @param blobInfo the blob information
+     * @param doc the document to which this blob belongs
+     * @param xpath the xpath of the blob in the document
+     * @return a managed blob
+     * @since 11.1
+     */
+    Blob readBlob(BlobInfo blobInfo, Document doc, String xpath) throws IOException;
```

Comment:
Why not using BlobInfoContext for the signature?

Reply:
For the `DocumentBlobManager` I kept simple explicit APIs. `BlobInfoContext` is currently used at lower level in `BlobProvider` to pass info around, but here I kept something symmetric with the existing `writeBlob(Blob blob, Document doc, String xpath)`. We could refactor this to use `BlobInfoContext` / `BlobContext` yes (although `BlobContext` doesn't currently keep the full doc, just its id). I'll keep that for later.